app/code/core/Mage/Catalog/Model/Resource/Category/Flat.php

```diff
@@ -1307,21 +1307,23 @@ public function isInRootCategoryList($category)
      * @param Mage_Catalog_Model_Category $category
      * @param bool $recursive
      * @param bool $isActive
+     * @param bool $sortByPosition
      * @return array
      */
-    public function getChildren($category, $recursive = true, $isActive = true)
+    public function getChildren($category, $recursive = true, $isActive = true, $sortByPosition = false)
     {
-        $maintable = $this->getMainStoreTable($category->getStoreId());
         $select = $this->_getReadAdapter()->select()
-            ->from($maintable, 'entity_id')
-            ->where('path LIKE ?', "{$category->getPath()}/%")
-            ->order($maintable.".position ASC");
```

Comment:
I expect the `$maintable` part is there for a specific reason. Without having a more in detail look on how this behaves in production I cant review this. Would be worried of an `column name is ambiguous` error because of some joins

Reply:
Please see collins change blow ...